File: ServerCore/src/main/java/org/elkoserver/foundation/server/RtcpConnectionSetup.java
package org.elkoserver.foundation.server;

import org.elkoserver.foundation.boot.BootProperties;
import org.elkoserver.foundation.net.MessageHandlerFactory;
import org.elkoserver.foundation.net.NetAddr;
import org.elkoserver.foundation.net.NetworkManager;
import org.elkoserver.foundation.server.metadata.AuthDesc;
import org.elkoserver.util.trace.Trace;

import java.io.IOException;

class RtcpConnectionSetup extends BaseTcpConnectionSetup {
    RtcpConnectionSetup(String label, String host, AuthDesc auth, boolean secure, BootProperties props, String propRoot, NetworkManager myNetworkManager, MessageHandlerFactory actorFactory, Trace trServer, Trace tr) {
        super(label, host, auth, secure, props, propRoot, myNetworkManager, actorFactory, trServer, tr);
    }

    @Override
    public String getProtocol() {
        return "rtcp";
    }

    @Override
    NetAddr createListenAddress() throws IOException {
        return myNetworkManager.listenRTCP(
                bind,
                actorFactory,
                msgTrace,
                secure);
    }
}

File: ServerCore/src/main/kotlin/org/elkoserver/foundation/server/RtcpConnectionSetup.kt
package org.elkoserver.foundation.server

import org.elkoserver.foundation.net.MessageHandlerFactory
import org.elkoserver.foundation.net.NetworkManager
import org.elkoserver.foundation.properties.ElkoProperties
import org.elkoserver.foundation.server.metadata.AuthDesc
import org.elkoserver.util.trace.TraceFactory
import org.elkoserver.util.trace.slf4j.Gorgel
import java.io.IOException

internal class RtcpConnectionSetup(
        label: String?,
        host: String,
        auth: AuthDesc,
        secure: Boolean,
        props: ElkoProperties,
        propRoot: String,
        myNetworkManager: NetworkManager,
        actorFactory: MessageHandlerFactory,
        gorgel: Gorgel,
        listenerGorgel: Gorgel,
        private val tcpConnectionGorgel: Gorgel,
        traceFactory: TraceFactory)
    : BaseTcpConnectionSetup(label, host, auth, secure, props, propRoot, myNetworkManager, actorFactory, gorgel, listenerGorgel, traceFactory) {
    override val protocol: String = "rtcp"

    @Throws(IOException::class)
    override fun createListenAddress() = myNetworkManager.listenRTCP(bind, actorFactory, listenerGorgel, msgTrace, tcpConnectionGorgel, secure)
}
